### DISCUSSION TITLE
Changing colors/fonts globally

### DISCUSSION THREAD

**Discussion by astrogewgaw** [lowdefy/lowdefy#754] [Questions and Help] [answered]
<div align="justify">

Is there a way to specify the colors and fonts that the **Lowdefy** app should use globally? Adding `style` attributes to individual components is a bit tiring, when what I actually want is to have the same colors/fonts used throughout. Normally, this is accomplished through custom CSS, but from what I understand, this is a feature that still has to be implemented? Is there a way to just do this for the colors and fonts, while leaving the rest of the settings unchanged?

</div>

**Comment by Gervwyk** [comment, marked as answer]
Hi @astrogewgaw 
Unfortunately, there is no official way at the moment. This can get a little complicated from an app dependency and build perspective. See discussion #678 

If you have any ideas on how to approach this, it would be appreciated. We'll definitely give custom antd theming another shot in the future ⚡️

**Reply by Gervwyk** [reply]
In some of our apps we set these style settings in `global` and then reference `_global` as required. It is not a great solution but it helps to make the style settings a bit more manageable.

**Reply by astrogewgaw** [reply]
<div align="justify">

@Gervwyk Ya I thought as such. No worries, I have something in the works that will take of this for my site at the very least. You can check it out if you wish in my website's `dev` branch. I am essentially using a Python-based templating mechanism to control all the YAML files now, so I can define global variables in a separate configuration file (also in YAML) and then let a script take care of the rest. It is all working beautifully 👍🏾 !

PS: I did think of using **Lowdefy's** inherent templating system, based on _Nunjucks_, but I wanted something that I could control dynamically, in a language I was comfortable with 😁 . Plus, I think you and @SamTolmay foresaw this as on of the most powerful ways to use **Lowdefy** anyways, right?

</div>

**Reply by SamTolmay** [reply]
I've been following your repo, and it is really interesting what you have been doing with Python. That is one of the things we are excited about, seeing what people can do by generating Lowdefy configuration.

We also have `resolver` and `transformer` functions on references that might be useful to you. But they do require you to write Node.js code.
